### ISSUE TITLE
multi-output keras models change metric names when reloading

### ISSUE THREAD

**Issue by aredier** [tensorflow/tensorflow#48692]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NO
- TensorFlow installed from (source or binary): pip install
- TensorFlow version (use command below): 2.4.1
- Python version: 3.79
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source): NA
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the current behavior**

When reloading a multi-output model, the name of the output gets repeated  when calling `tf.keras.models.Model.evaluate`:

```
>>> import tensorflow as tf
>>> tf.__version__
'2.4.1'

>>> input_a = tf.keras.Input(shape=(1,))

>>> ifoo = tf.keras.layers.Dense(1, name="foo")(input_a)
>>> ibar = tf.keras.layers.Dense(1, name="bar")(input_a)

>>> imodel = tf.keras.Model(input_a, {"foo": foo, "bar": bar})
>>> imodel.compile(loss="mse", optimizer="adam", metrics={"foo": "mae", "bar": "mae"})
>>> model.evaluate(tf.constant([[1]]), tf.constant([[1]]), return_dict=True)

1/1 [==============================] - 0s 236ms/step - loss: 4.0179 - bar_loss: 3.9153 - foo_loss: 0.1026 - bar_mae: 1.9787 - foo_mae: 0.3203
{'loss': 4.0179123878479,
 'bar_loss': 3.915294885635376,
 'foo_loss': 0.10261743515729904,
 'bar_mae': 1.978710412979126,
 'foo_mae': 0.320339560508728}

>>> model.save("/tmp/model")
>>> model_relaoded = tf.keras.models.load_model("/tmp/model")
>>> model_relaoded.evaluate(tf.constant([[1]]), tf.constant([[1]]), return_dict=True)

1/1 [==============================] - 0s 87ms/step - loss: 4.0179 - bar_loss: 3.9153 - foo_loss: 0.1026 - bar_bar_mae: 1.9787 - foo_foo_mae: 0.3203
{'loss': 4.0179123878479,
 'bar_loss': 3.915294885635376,
 'foo_loss': 0.10261743515729904,
 'bar_bar_mae': 1.978710412979126,
 'foo_foo_mae': 0.320339560508728}
```

**Describe the expected behavior**
I would expect the metrics' name to remain unchanged when reloading:

```
>>> model.save("/tmp/model")
>>> model_relaoded = tf.keras.models.load_model("/tmp/model")
>>> model_relaoded.evaluate(tf.constant([[1]]), tf.constant([[1]]), return_dict=True)

1/1 [==============================] - 0s 87ms/step - loss: 4.0179 - bar_loss: 3.9153 - foo_loss: 0.1026 - bar_mae: 1.9787 - foo_mae: 0.3203
{'loss': 4.0179123878479,
 'bar_loss': 3.915294885635376,
 'foo_loss': 0.10261743515729904,
 'bar_mae': 1.978710412979126,
 'foo_mae': 0.320339560508728}
```

**Standalone code to reproduce the issue**

Here is a collab with the above code: https://colab.research.google.com/drive/1rm2d51gXw0ZvxIs63Qu5ACyRDK5OAgoM?usp=sharing

**Other info / logs** Include any logs or source code that would be helpful to
NA

PS: I have never contributed to TensorFlow but I would be glad to help resolve this issue.

**Comment by UsharaniPagadala** [comment]
@aredier 
 I reproduced the code in tf-nightly it worked as you expected that the metrics' remain unchanged when reloading:
Could you please find the [gist](https://colab.research.google.com/gist/UsharaniPagadala/c43c04782946b15d3faf7a9a671bde9d/-48692.ipynb) here.Thanks

**Comment by aredier** [comment]
@UsharaniPagadala I can indeed reproduce on my machine with tf_nightly.
Tnanks
